packages/dataviews/src/normalize-fields.ts

```diff
@@ -21,6 +21,27 @@ const getValueFromId =
 		return value;
 	};
 
+// This is a fake schema parser that only supports minLength and maxLength
+const fakeSchemaParser = (
+	schema: Record< 'minLength' | 'maxLength', number >
+) => {
+	const callbacks = {} as {
+		[ key: string ]: ( value: any ) => string | undefined;
+	};
+
+	if ( schema.minLength !== undefined ) {
+		callbacks.minLength = ( value ) =>
+			value.length < schema.minLength ? 'Length is too short' : undefined;
+	}
+
+	if ( schema.maxLength !== undefined ) {
+		callbacks.maxLength = ( value ) =>
+			value.length >= schema.maxLength ? 'Length is too long' : undefined;
+	}
+
+	return callbacks;
```

Comment:
It is just a POC

Reply:
This should be part of each "Field type" somehow, each field type has its own "config/constraints"

Reply:
Still not ready for another review!

Reply:
We have field types "number", "text" ... and each field type should define a validation logic for it.

 - number should support "min"/"max" constraints".
 - string should support "minLenght"/"maxLength".
 - isRequired is a constraint that works for all field types.

What I'm saying is that I feel this logic should be provided by the field types and not some global function like this.

Reply:
Yeah, we should update the existing `isValid` method of each field type to take into account these new properties (min/max, etc.) ([example](https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/field-types/text.tsx#L13)).